pkg/operator/events/recorder.go

```diff
@@ -40,14 +41,35 @@ func GetControllerReferenceForCurrentPod(client corev1client.PodInterface) (*cor
 	if err != nil {
 		return nil, err
 	}
-	ownerRef := metav1.GetControllerOf(pod)
+	// get replica set owner reference from pod, if we fail use the pod itself as involved object
+	replicaSetOwnerRef := metav1.GetControllerOf(pod)
+	if replicaSetOwnerRef == nil || replicaSetOwnerRef.Kind != "ReplicaSet" {
+		return makeObjectReference(&metav1.OwnerReference{Kind: "Pod", Name: pod.Name, UID: pod.UID, APIVersion: "v1"}, pod.Namespace), nil
```

Comment:
so if the pod were owned by a daemonset, this would return an indication that it was linked to a pod?

Perhaps it will be easier to express as a recursive function that takes an object reference as input and returns the most root owner.  Each stage calling back into itself.  A switch case for handled types.